oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -75,6 +81,18 @@ struct NodeInfo {
     join_handle: Option<JoinHandle<()>>,
 }
 
+/// The downgrading (declassification + endorsement) privilege associated with a Node instance.
+///
+/// See https://github.com/project-oak/oak/blob/master/docs/concepts.md#downgrades
+#[derive(Debug, Default, Clone)]
+pub struct NodePrivilege {
+    /// Tags that may be declassified (removed from the secrecy component of a label) by the Node.
+    can_declassify_secrecy_tags: HashSet<Tag>,
+
+    /// Tags that may be endorsed (added to the integrity component of a label) by the Node.
+    can_endorse_integrity_tags: HashSet<Tag>,
+}
+
```

Comment:
@aferr I'm encapsulating the capabilities (of both kinds) into this struct for now, is "privilege" a correct / good enough term? I am trying to avoid "capability" because it may have other implications in the context of operating systems and handles to channels etc.

FWIW I am still hoping eventually to unify this with the hierarchy of principals a la Flam, but that's future work :)

Reply:
Yeah looks good I like the way you did it.

Not using the word capability does make sense and privilege is a good alternative!

We might want something easier than FLAM to explain an understand but has the benefits of FLAM.One tricky part about making a "simpler FLAM" might be that (I think) having two separate objects for "integrity" and "declassifying privilege" is easier to explain than when they are the same thing, but suspect (but am not certain) separating them will have implications w.r.t. what theorems we can prove. Also something to consider as part of that future work.